azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -1384,4 +1665,161 @@ private void cleanExecutionLogs() {
           - executionLogsRetentionMs);
     }
   }
-}
+
+  /*
+   * This thread is responsible for processing queued flows.
+   */
+  private class QueueProcessorThread extends Thread {
+    private static final long QUEUE_PROCESSOR_WAIT_IN_MS = 1000;
+    private static final int MAX_DISPATCHING_ERRORS_PERMITTED = 5;
+    private final long activeExecutorRefreshWindowInMilisec;
+    private final int activeExecutorRefreshWindowInFlows;
+
+    private boolean shutdown = false;
+    private boolean isActive = true;
```

Comment:
I believe this needs to be volatile, because it will be updated and read by different threads.